contracts/INKPurchase.sol

```diff
@@ -9,22 +9,24 @@ contract INKPurchase is Ownable {
 	using SafeERC20 for IERC20;
 
 	address public immutable usdc;
+	address public treasuryWallet;
 	address public constant weth9 =
-		0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+		0xc778417E063141139Fce010982780140Aa0cD5Ab;
 	ISwapRouter public uniswapRouter;
 	mapping(address => uint256) public purchasedAmount;
 
 	event Purchased(address buyer, uint256 amount);
 
-	constructor(address _routerAddress, address _usdc) {
-		uniswapRouter = ISwapRouter(_routerAddress);
+	constructor(address _treasuryWallet, address _usdc) {
+		treasuryWallet = _treasuryWallet;
+		uniswapRouter = ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
```

Comment:
`swapRouter` address should be a variable and should be initialized in constructor.

Reply:
okay, swap router is unique for all chains.